R/scale-.r

```diff
@@ -184,6 +184,7 @@ discrete_scale <- function(aesthetics, scale_name, palette, name = waiver(),
 
   check_breaks_labels(breaks, labels)
 
+  if (is.formula(labels)) labels <- as_function(labels)
```

Comment:
Sorry this is mostly my fault for getting in your way with #4427, but the `allow_lambda()` statements below already capture formulas as functions, so this line has become unnecessary. Apologies!